src/txmempool.h

```diff
@@ -883,47 +867,49 @@ struct DisconnectedBlockTransactions {
     // reorg, besides draining this object).
     ~DisconnectedBlockTransactions() { assert(queuedTx.empty()); }
 
-    indexed_disconnected_transactions queuedTx;
-    uint64_t cachedInnerUsage = 0;
-
-    // Estimate the overhead of queuedTx to be 6 pointers + an allocation, as
-    // no exact formula for boost::multi_index_contained is implemented.
     size_t DynamicMemoryUsage() const {
-        return memusage::MallocUsage(sizeof(CTransactionRef) + 6 * sizeof(void*)) * queuedTx.size() + cachedInnerUsage;
+        // std::list has 3 pointers per entry
+        return cachedInnerUsage + memusage::DynamicUsage(iters_by_txid) + memusage::MallocUsage(3 * sizeof(void*)) * queuedTx.size();
     }
 
     void addTransaction(const CTransactionRef& tx)
```

Comment:
Hmm, why no `addTransactions(const std::vector<CTransactionRef>& vtx)` to match `removeForBlock()` ?

Is it because the tx order needs to be reversed for some reason that's not mentioned? :p

That would allow a `iters_by_txid.reserve(vtx.size());` which gets me a ~10% speedup in the benches.

Reply:
> Hmm, why no addTransactions(const std::vector<CTransactionRef>& vtx) to match removeForBlock() ?

I had been trying to keep the diff as small as possible

> That would allow a iters_by_txid.reserve(vtx.size()); which gets me a ~10% speedup in the benches.

This is worth it though, so I've done this now - thanks!

> Edit: removed snark after finding the comment I needed.

Added a comment as I agree it wasn't super obvious why we reverse vtx